src/System.CommandLine.StaticCompletions/shells/BashShellProvider.cs

```diff
@@ -0,0 +1,222 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+namespace System.CommandLine.StaticCompletions.Shells;
+
+using System.CodeDom.Compiler;
+using System.CommandLine;
+using System.CommandLine.Completions;
+using System.IO;
+
+public class BashShellProvider : IShellProvider
+{
+    public string ArgumentName => "bash";
+
+    public string Extension => "sh";
+
+    public string GenerateCompletions(CliCommand command)
+    {
+        var initialFunctionName = command.FunctionName().MakeSafeFunctionName();
+        return
+            $"""
+            #! /bin/bash
```

Comment:
The "shebang" needs to be on the first line of the file, not the second. I validated with a script that does `echo "$BASH_VERSION"` and when there was an empty line before `#!/usr/bin/bash`, then running the script from a shell that does not define `BASH_VERSION`, then the output was empty. Removing the empty line from the script made it work.

Reply:
@KalleOlaviNiemitalo is correct - you can see the generated completions in the [snapshot](https://github.com/baronfel/sdk/blob/427f24df7f89933f0c7257f4ad377f9e26eeea1d/test/dotnet-completions.Tests/snapshots/bash/DotnetCliSnapshotTests.VerifyCompletions.verified.sh#L19) for bash, and it looks well-formed in this regard.

Reply:
How did I not know about [this](https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/tokens/raw-string)?  I'm definitely going to be using it a lot going forward!  It'll be amazing for a bunch of NuGet's test code.